frontend/components/ManageStakingPage/StakingContractSection/MigrateButton.tsx

```diff
@@ -128,23 +128,48 @@ export const MigrateButton = ({
             overrideSelectedServiceStatus(MiddlewareDeploymentStatus.DEPLOYING);
             goto(Pages.Main);
 
-            const serviceConfigParams = {
-              stakingProgramId: stakingProgramIdToMigrateTo,
-              serviceTemplate,
-              deploy: true,
-              useMechMarketplace:
-                stakingProgramIdToMigrateTo ===
-                StakingProgramId.PearlBetaMechMarketplace,
-            };
-
             if (selectedService) {
               // update service
               await ServicesService.updateService({
-                ...serviceConfigParams,
                 serviceConfigId,
+                partialService: {
+                  chain_configs: {
```

Comment:
@OjusWiZard looking at this code https://github.com/valory-xyz/olas-operate-app/pull/621/files#diff-459f04e9c6b0735617705b19c529ccf7d3ab8e4ca3d024c74f62529ed82ae23bR1037-R1039 I can't understand should I pass the whole `chain_configs` just in case, or I can pass only `user_params` (following the same structure) with only `staking_program_id` and `use_mech_marketplace`

Reply:
by looking at that code link, only passing the required key-values should be enough. But did you test it that way?

Reply:
I did and it didn't work 🫠

Reply:
now it works